### ISSUE TITLE
INWX: Does not work with more than 20 Domains

### ISSUE THREAD

**Issue by schwarmco** [StackExchange/dnscontrol#2512]
**Describe the bug**

Following the Quickstart Guide i've ran into the error:

```bash
$ dnscontrol preview
[INFO: Diff2 algorithm in use. Welcome to the future!]
******************** Domain: mydomain.com
WARNING: Zone 'mydomain.com' does not exist in the 'inwx' profile and will be added automatically.
```

After a lot of debugging i finally figured, that `dnscontrol` only works with the first 20 domains in my account.

This is because the `goinwx` client or the INWX-API in general has paginated results, but the INWX-Provider in dnscontrol only fetches the first page.

See: 
https://github.com/StackExchange/dnscontrol/blob/194c4711a89175cb0ba5e14882682bbc5fe07ff1/providers/inwx/inwxProvider.go#L403-L416
and: https://github.com/nrdcg/goinwx/blob/073f789977c3777041908f6a9990ebd1fdc356a1/domain.go#L329 

**To Reproduce**
1. Have more than 20 Domains in INWX
2. Use this dnsconfig.js:
```javascript
var REG_INWX = NewRegistrar("inwx");
var DNS_INWX = NewDnsProvider("inwx");

D("mydomain.com", REG_INWX, DnsProvider(DNS_INWX), 
    A('@', '1.1.1.1')
);
```
3. mydomain.com has to not be in the first 20 results (domains sorted alphabetically)
5. See error message above

**Expected behavior**
All domains should be recursively fetched from INWX.

**DNS Provider**
 - INWX

**Additional context**
I initially thought, the problem was with omitting `DnsProvider(DNS_INWX)` in the `dnsconfig.js` (see closed #2511) but this just prevented the missing records to throw an error - sorry for the confusion.

**Comment by tlimoncelli** [comment]
CC @patschi (provider maintainer for INWX)

**Comment by synotna** [comment]
FYI this broke in 4.2.0

Going back down to 4.1.1 is a workaround until fixed

**Comment by patschi** [comment]
That's interesting...

The only two changes [as per history](https://github.com/StackExchange/dnscontrol/commits/master/providers/inwx) are: #2428 and #2430. Both of them were not changing anything in regards to the `fetchNameserverDomains()` function.

Also the [last change with the goinwx dependency](https://github.com/StackExchange/dnscontrol/blame/master/go.mod#L39) is more than 7 months ago - way before dnscontrol 4.1.1 was released.

@tlimoncelli Do you know if `ListZones()` is used in some special form? Because that's the only real change.

I still need to find some time reproducing and debugging, hopefully on the weekend. I should be able to create a few domains on their sandbox.

**Comment by schwarmco** [comment]
No - i still can't access domains past position 20 in INWX provider... i still believe, its because the list of domains does not get fetched paginated:

https://github.com/StackExchange/dnscontrol/blob/08afef7f7c2c34674525f729f1b41edb06aa1667/providers/inwx/inwxProvider.go#L403-L416

unlike in the dnsimple provider for example:

https://github.com/StackExchange/dnscontrol/blob/08afef7f7c2c34674525f729f1b41edb06aa1667/providers/dnsimple/dnsimpleProvider.go#L596-L614

**Comment by patschi** [comment]
Got much going on here. I'll try to get to it this weekend. Sorry!

@tlimoncelli 
> Can this be closed?

Have you got some input on https://github.com/StackExchange/dnscontrol/issues/2512#issuecomment-1690388350? Because Joachim mentions it is working in 4.1.1, but not 4.2.0 and later.

**Comment by schwarmco** [comment]
this actually wasn't my comment :) 

i haven't tested 4.1.1 and now cannot confirm that it worked because of the recent breaking-change (see https://github.com/StackExchange/dnscontrol/issues/2550)

**Comment by patschi** [comment]
> this actually wasn't my comment :)

Sorry, indeed. Mixed you up. I don't think that older versions work here, as it seems to be indeed related to paging present from INWX API even before.

I have proposed a simple fix in PR #2566 right now. Tested and working fine.

**Comment by tlimoncelli** [comment]
@patschi Would you please submit a PR that documents this?  I think anyone that reads `documentation/providers/inwx.md` should be forewarned that the provider will fail if they have more than 2,147,483,647 domains.
